README.md

```diff
@@ -87,10 +87,40 @@ app = FastAPI()
 app.include_router(example)
 
 ```
+
+
+
 ## Required parameters
 - db_model: The SqlAlchemy model you want to create CRUD endpoints for
 - db_func: The function that returns the SqlAlchemy session you'd like to use
 
+
+## Database session handling
+The `db_func` parameter is expected to be a function that returns a SQLAlchemy session. This function will be injected using FastAPI's `Depends` function.
+The generated endpoint will handle adding and committing the session, as well as rolling back the session in the event of an error.
+
+This follows the FastAPI docs on dependency injection. see: https://fastapi.tiangolo.com/tutorial/dependencies/ and https://fastapi.tiangolo.com/reference/dependencies/
```

Comment:
would use this link instead https://fastapi.tiangolo.com/tutorial/sql-databases/ since it uses the same `get_db` pattern we're defining here